polyply/src/random_walk.py

```diff
@@ -65,54 +66,22 @@ def _is_overlap(meta_molecule, point, tol, fudge=1):
     -------
     bool
     """
-    for node in meta_molecule:
-        try:
-            coord = meta_molecule.nodes[node]["position"]
-        except KeyError:
-            continue
-
-        if np.linalg.norm(coord - point) < tol * fudge:
-            return True
-
+    red_pos = positions[positions[:, 0] != np.inf]
+    traj_tree = scipy.spatial.ckdtree.cKDTree(red_pos)
+    ref_tree = scipy.spatial.ckdtree.cKDTree(point.reshape(1,3))
+    dist_mat = ref_tree.sparse_distance_matrix(traj_tree, 1.1)
+    red_types = atom_types[positions[:, 0] != np.inf ]
+
+    current_atom = atom_types[gndx]
+    for pair, dist in dist_mat.items():
+        #print(dist)
+        ref = vdw_radii[frozenset([current_atom, red_types[pair[1]]])]
+        if dist < ref*0.8:
+           return True
     return False
 
-def update_positions(vector_bundle, meta_molecule, current_node, prev_node):
-    """
-    Take an array of unit vectors `vector_bundle` and generate the coordinates
-    for `current_node` by adding a random vector to the position of the previous
-    node `prev_node`. The length of that vector is defined as 2 times the vdw-radius
-    of the two nodes. The position is updated in place.
-
-    Parameters
-    ----------
-    vector_bunde: np.ndarray(m,3)
-    meta_molecule: :class:polyply.src.meta_molecule.MetaMolecule
-    current_node: node_key[int, str]
-    prev_node: node_key[int, str]
-    """
-    if "position" in meta_molecule.nodes[current_node]:
-        return
-
-    current_vectors = vector_bundle.copy()
-    last_point = meta_molecule.nodes[prev_node]["position"]
-
-    prev_resname = meta_molecule.nodes[prev_node]["resname"]
-    current_resname = meta_molecule.nodes[current_node]["resname"]
-
-    current_vdwr = meta_molecule.volumes[current_resname]
-    prev_vdwr = meta_molecule.volumes[prev_resname]
-    vdw_radius, _ = lorentz_berthelot_rule(current_vdwr, prev_vdwr, 1, 1)
-
-    # we give 10 percent more than the vdw radius so we don't generate a
-    # self ovelap
-    step_length = 1.1*vdw_radius
-    while True:
-        new_point, index = _take_step(vector_bundle, step_length, last_point)
-        if not _is_overlap(meta_molecule, new_point, tol=vdw_radius):
-            meta_molecule.nodes[current_node]["position"] = new_point
-            break
-        else:
-            vector_bundle = np.delete(vector_bundle, index, axis=0)
+def not_exceeds_max_dimensions(point, maxdim):
+    return np.all(point < maxdim) and np.all(point > np.array([0., 0., 0.]))
```

Comment:
```suggestion
    return np.all((point <= maxdim) & (point >= 0))
```
You could also try:
`np.all(0 < point < maxdim)`
but that probably doesn't work

Reply:
works as it is now.